File: 0.test.cpp
/*
 * @Description: 
 * @Version: 1.0
 * @Autor: Vicro
 * @Date: 2020-12-14 20:07:31
 * @LastEditTime: 2020-12-14 21:24:42
 * @FilePath: \Leetcode\0.test.cpp
 */
/*
 * @lc app=leetcode.cn id=49 lang=cpp
 *
 * [49] 字母异位词分组
 */

// @lc code=start
#include <iostream>
#include <vector>
#include <unordered_map>
#include <algorithm>
using namespace std;


/*
RESULT: Accept
TIME:   ms    BEAT %    O(n) = 
MEMORY: MB    BEAT %    O(n) = 
Description: 
*/

class Solution {
public:
    vector<vector<string>> groupAnagrams(vector<string> strs) {
        unordered_map<string, vector<string>> hash_map;
        
        for (auto s: strs) {
            string temp = s;
            sort(temp.begin(), temp.end());
            hash_map[temp].push_back(s);
        }

        int len = hash_map.size();
        vector<vector<string>> ans(len);
        int index = 0;
        for (auto s: hash_map) {
            ans[index] = s.second;
            ++ index;
        }

        return ans;
    }
};


int main() {
    Solution sol;
    int A = 0;
    int B = 0;
    vector<vector<string>> ans = sol.groupAnagrams({"eat", "tea", "tan", "ate", "nat", "bat"});
    for (int i = 0; i < ans.size(); i ++) {
        cout << ans[i] << endl;
    }
    system("pause");
    return 0;
}

File: 0.test.py
'''
Description: k
Version: 1.0
Author: Vicro
Date: 2020-12-02 13:45:44
LastEditTime: 2020-12-02 15:34:52
FilePath: \Leetcode\0.test.py
'''
#
# @lc app=leetcode.cn id=321 lang=python3
#
# [321] 拼接最大数
#

# @lc code=start

A = [1, 3, 2]

A = map(A)

B = 1
class Solution:
    def maxNumber(self, nums1, nums2, k):
        
        def choose_max(nums, k, list, res):
            if len(list) == temp:
                res.append(list)
                return res
            for j in range(len(nums) - k + 1):
                # print(nums[len(nums) - k])
                res = choose_max(nums[j + 1:], k - 1, list + [nums[j]], res)
            return res

        for i in range(k + 1):
            if i <= len(nums1) and k - i <= len(nums2):
                temp = i
                res1 = choose_max(nums1, i, [], [])
                temp = k - i
                res2 = choose_max(nums2, k - i, [], [])
                A = 1

import 
sol = Solution()
sol.maxNumber([3, 4, 6, 5], [9, 1, 2, 5, 8, 3], 5)
                
        
# @lc code=end

